### ISSUE TITLE
Snagit Key

### ISSUE THREAD

**Issue by marphlix** [autopkg/bkerns-recipes#29]
Can someone please help me with the autopkg and applying the snagit key via JAMF. I am very new to this, and don't know much about anything. I can create the package with autopkg fine, but i am not sure how to apply the reg key with JAMF as per:
_To use SNAGIT_KEY, create an override and add the key value. The needed key value can be pulled from a machine that has already been registered with the following:_ 

cp /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey.plist
defaults read /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey.plist RegKey | sed 's/[<> ]//g'*

I tried editing the override file to add the key provided by snagit it to : here:
`<dict> <key>NAME</key> <string>TechSmithSnagit</string> <key>SNAGIT_KEY</key> <string>xxx</string>`

But that didn't generate the Snagitregistrationkey in the /Users/Shared dir when i installed the package.

The command cp /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey.plist
defaults read /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey.plist RegKey | sed 's/[<> ]//g' shows **{length=60, bytes=0xfffxxxx...30434}** and not sure how that then needs to be applied to the overwrite file.

Any help would be greatly appreciated.

**Comment by MLBZ521** [comment]
@marphlix So, the process that Brandon setup, works like this:

1. You would install and license Snagit on a machine
2. From there, you would run the commands that you mentioned to pull the license key
3. The output from the second command would be added to your recipe override, set to the `SNAGIT_KEY` Input Variable

Regarding the commands:
```
cp /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey.plist
defaults read /Users/Shared/TechSmith/Snagit/SnagitRegistrationKey.plist RegKey | sed 's/[<> ]//g'
```
- The first command creates a copy of the file and gives it a .plist extension so the defaults command can read it.
- The second command just performs some character subsitution.



However, what I'm doing is providing my own Scripts folder along with my recipe override so that I can _override_ what Brandon has in his scripts.  Within the postinstall script, I have the output from the second command set as the value of the `regkey` variable.

Hope this helps.

**Comment by WaveBirdDash** [comment]
Worth noting that Snagit 2021 has changed the license key injection method to match Camtasia 2020+, so the plist is no longer the correct file path. #31 updates the postinstall script to account for this change.

**Comment by MLBZ521** [comment]
I was provided the 2021 license key right after my post and realized that the licensing method changed as well.

Finally found your info in #25 with a link on how to apply the license for Camtasia.  Looks like TechSmith did the same thing again, releasing the ware without updating the documentation.

I went ahead and submitted a PR (#32) to update the README's documentation for the TechSmith wares.  Hopefully that'll help anyone with future questions.

**Comment by kernsb** [comment]
Resolved by pull request #31

**Comment by WaveBirdDash** [comment]
See you all in a year or so when they change the workflow again!
